src/nvmsgbroker_pipeline.py

```
@@ -0,0 +1,147 @@
+
+def direct_mqtt_probe(pad, info, u_data):
+    gst_buffer = info.get_buffer()
+    batch_meta = pyds.gst_buffer_get_nvds_batch_meta(hash(gst_buffer))
+    
+    l_frame = batch_meta.frame_meta_list
+    while l_frame is not None:
+        frame_meta = pyds.NvDsFrameMeta.cast(l_frame.data)
+
+        # create payload
+        payload = {
+            "timestamp": datetime.now().isoformat(),
+            "frame_number": frame_meta.frame_num,
+            "source_id": frame_meta.source_id,
+            "pipeline": "nvmsgbroker_direct",
+            "objects_detected": 1
+        }
+
+        #   send to MQTT
+        try:
+            client = mqtt.Client()
+            client.connect(MQTT_BROKER, MQTT_PORT, 60)
+            client.publish(MQTT_TOPIC, json.dumps(payload))
+            client.disconnect()
+            print(f" Sent frame {frame_meta.frame_num} to MQTT")
+        except Exception as e:
+            print(f" MQTT Error: {e}")
+        
+        l_frame = l_frame.next
+    
+    return Gst.PadProbeReturn.OK
+
+def build_pipeline():
+    pipeline = Gst.Pipeline.new("direct-mqtt-pipeline")
+
+    # Source elements
+    rtspsrc = Gst.ElementFactory.make("rtspsrc", "source")
+    depay = Gst.ElementFactory.make("rtph264depay", "depay")
+    parse = Gst.ElementFactory.make("h264parse", "parse")
+    decode = Gst.ElementFactory.make("decodebin", "decode")
+    convert = Gst.ElementFactory.make("videoconvert", "convert")
+    nvvideoconvert = Gst.ElementFactory.make("nvvideoconvert", "nvvideoconvert")
+    
+    # DeepStream elements
+    streammux = Gst.ElementFactory.make("nvstreammux", "streammux")
+    nvinfer = Gst.ElementFactory.make("nvinfer", "nvinfer")
+    fakesink = Gst.ElementFactory.make("fakesink", "sink")
+    
+    # Add elements to pipeline
+    elements = [rtspsrc, depay, parse, decode, convert, nvvideoconvert, 
+                streammux, nvinfer, fakesink]
+    for element in elements:
+        pipeline.add(element)
```

Creating GStreamer elements without validating their successful initialization may result in silent failures or unexpected pipeline behavior, especially if a required plugin is missing.

Therefore, each `Gst.ElementFactory.make(...)` call should be followed by a check to ensure the element is not `None`, and proper error handling should be added if creation fails.